.github/workflows/validate.yml

```diff
@@ -0,0 +1,22 @@
+name: validate-json
+
+on:
+  [push]
+  # push:
+  #   branches: ["main"]
+  # pull_request:
+  #   branches: ["main"]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: ">=1.22.4"
+      - name: Install dependencies
+        run: go install github.com/santhosh-tekuri/jsonschema/cmd/jv@latest
+      - name: Validate JSON files
+        run: jv test/data/snapshot-schema.json test/data/json/*.json
```

Comment:
this just tests our validation right, it doesn't validate the generated data from the yaml?

Reply:
Ideally, I think it should do both.

After I [add steps](https://github.com/thunderbird/thunderbird-notifications/pull/9#issuecomment-2187519769) to save the generated json from #9, I can update the `jv` command so it validates the generated json as well.

(i.e., change the path from `test/data/json/*.json` to `json/*.json`)

Reply:
tests to make sure the validation of our test json should happen in a different job, one that runs against every PR against this codebase to make sure we don't break the main functionality of this repo. Otherwise, if we land https://github.com/thunderbird/thunderbird-notifications/pull/9/files, then we will potentially merge yaml files that fail json validation. That's not ideal, and the typical DevOps/CI/what-have-you way of handling our workflows would be:

- have a job that runs every PR to ensure we don't break what we're building. In our case, this job would verify that the json we currently have in the repo is valid, and this runs on every PR.
- When a user is submitting new content (yaml in this case), we should have a job that converts and then runs the previous job to ensure all the json is valid before we commit it to the repo. In this case, it would be when the yaml/ folder is updated, we run the yaml->json converter, and then we run the jv validator on the output, then we commit it. If any of those steps fail, nothing gets committed

Reply:
Sounds good!
To answer your original question: yes, this just validates the JSON we already have.

And I totally see your point about having separate jobs:
1. this one, which validates the json files we currently have and runs automatically on every PR
2. a second one that converts yaml to json, then:
  a. runs job 1 to validate json (including the new json)
  b. only if validation job passes, commit the new json to repo
 
I'll update the path from `test/data/json/*.json` to `json/*.json` and then merge this PR (so that #9 can be updated to run the validation after conversion)